README.md

```diff
@@ -1,13 +1,132 @@
+
+<details>
+ <summary>Installing GNU requirements on Debian/Ubuntu</summary>
+
+On Debian/Ubuntu the GNU requirements can be met by
+
+```
+apt install gcc g++ git zlibc zlib1g-dev libssl-dev wget make
+```
+
+</details>
+
+## Installation and setup
+
+1. Clone the repo
+
+<details>
+ <summary>Note on cloning on LXPLUS</summary>
+
+When trying to use SSH with GitHub on LXPLUS it is important to make sure that your `~/.ssh/config` is properly configured. It may need to contain something along the lines of
+
+```
+Host github.com
+    IdentityFile ~/.ssh/id_rsa-github
+    IdentitiesOnly yes
+```
+
+</details>
+
+
+2. If the above requirements are not already satisfied, run the installer
+
+```
+bash scripts/install.sh
+```
+
+3. Make a directory for your project
+
+```
+mkdir ~/projects/HEPML_project
+cd ~/projects/HEPML_project
+```
+
+4. Copy the included `Pipfile` and `Pipfile.lock` to the project directory
+
+5. Install the environment with pipenv
+
+```
+# From inside the project directory
+pipenv install
+```
+
+6. Launch into the project environment with pipenv
+
+```
+pipenv shell
+```
+
+## Questions
+
+### Why were the packages chosen for the default environment?
+
+NumPy, SciPy, Matplotlib, and Pandas represent the core of the powerful [SciPy stack](https://www.scipy.org/) (IPython is taken care of by Jupyter). h5py and uproot allow for working with [industry standard data formats](https://support.hdfgroup.org/HDF5/). scikit-learn, scikit-image, and tensorflow provide a very strong machine learning ecosystem and Keras support. Jupyter provides the Jupyter notebook, qtconsole, and the IPython kernel, which ties it all together for interactive computing and data exploration.
+
+### Why not use Conda?
+
+This can also be done with Conda. However, sometimes Conda lags behind in releases and there are some benefits from using packages straight from PyPI as the developer's CD release them. In addition, Conda requires additional space and software that is not core to the environment that we need. This is not necessarily a problem, but having a lightweight, portable, reproducible environment that can be fully installed in under 5 minutes is the goal here, and I think that pipenv is better suited for that.
+
+However, it is recognized that Conda is incredibly popular, and even the recommended installation method for some packages. In the future a Conda `environment.yml` file that closely follows the `Pipfile` for the default environment will be released on [Anaconda cloud](https://anaconda.org/).
+
+### Why is ROOT not included?
+
+ROOT is not a necessary component of a HEP focused machine learning workflow. Any I/O involving `.root` files is handled by [uproot](https://github.com/scikit-hep/uproot), allowing you to focus on the work and not the file format.
+
+### Why is Keras not included?
+
+It is. [Keras is part of TensorFlow](https://www.tensorflow.org/api_docs/python/tf/keras) as of circa January, 2017.
```

Comment:
I'm a bit ambivalent here: on one hand, using the TF implementation of Keras simplifies things for users, which is always a plus. On the other hand, Keras does support multiple backends, and I don't want to discourage people from exploring them. True, at this point it's only three, and theano is no longer being developed, but I wouldn't want to discourage people from trying out [CNTK][1] (somewhere I read that their RNNs were faster than TF in some cases).


[1]: https://github.com/Microsoft/CNTK

Reply:
@dguest I could add Keras as a `dev` package (c.f. https://github.com/matthewfeickert/HEPML-env/blob/master/Pipfile#L18). What are your thoughts on that?

Reply:
It's a different module path, no? So to have other people's code work out of the box we have to choose one or the other. I'm partial to installing an extra package (it should be much lighter than TF) rather than going full tensorflow on this.

Reply:
Ah, I think I see what you're saying. Given that people are used to using Keras as its own library in the past then if people are going to want to have their own old code work then it should be included. This is a very good point.

Question time: If we want to shape the behavior of how people code should Keras be included in the default? Or should it be a dev package such that if people can't find it and read the questions part of the `README` they are encouraged to run

```
pipenv install --dev
```

and then they're good to go?

Reply:
I think it should be a default package. Honestly I think there will be more people who want to use this than there are who will use sk-image.

Reply:
> [`tf.keras` is TensorFlow's implementation of the Keras API specification](https://www.tensorflow.org/guide/keras#import_tfkeras)

Yeah, since I now realize `tf.keras` is different then the Keras library I am in full agreement with you. This will get change in an upcoming PR.

Reply:
Added to the `README` in commit 99d4f763b920e8e413262bda04cece0313b45799